File: sem2/po/po_proj2/src/studia/prog_ob/projekt2/world_simulator/organisms/plants/plants_subclasses/Dandellion.java
package studia.prog_ob.projekt2.world_simulator.organisms.plants.plants_subclasses;

import studia.prog_ob.projekt2.world_simulator.World;
import studia.prog_ob.projekt2.world_simulator.maps.MapLocation;
import studia.prog_ob.projekt2.world_simulator.organisms.Organism;
import studia.prog_ob.projekt2.world_simulator.organisms.plants.Plant;

public class Dandellion extends Plant {
    public static final int DANDELLION_SEEDING_THRESHOLD = BASE_PLANT_SEEDING_THRESHOLD;
    public static final int DANDELLION_STRENGTH = BASE_PLANT_STRENGTH;
    public static final int DANDELLION_INITIATIVE = BASE_PLANT_INITIATIVE;
    public static final char DANDELLION_SYMBOL = 'd';
    public static final int DANDELLION_SEEDING_ATTEMPTS_PER_TURN = 3;


    public Dandellion(MapLocation new_map_location, World new_world) {
        super(DANDELLION_STRENGTH, DANDELLION_INITIATIVE, new_map_location, DANDELLION_SYMBOL, new_world, DANDELLION_SEEDING_THRESHOLD);
    }

    @Override
    public Organism spawn_offspring(MapLocation location) {
        return new Dandellion(location, getWorld());
    }

    @Override
    public void action() {
        for (int i = 0; i < DANDELLION_SEEDING_ATTEMPTS_PER_TURN; i++) {
            super.seed();
        }
    }

    @Override
    public String toString() {
        return "Dandellion " + getMap_location() + " ";
    }
}

File: sem2/po/po_proj1/Plants/Plants_subclasses/Dandellion.cpp
//
// Created by M Sz on 09/04/2023.
//

#include "Dandellion.h"

std::string Dandellion::login()
{
    return std::string("Dandellion");
}

Dandellion::Dandellion(int new_x, int new_y, World &new_world)
        : Plant(DANDELLION_STRENGTH, new_x, new_y, DANDELLION_SYMBOL, DANDELLION_SEEDING_THRESHOLD, new_world)
{}

Plant *Dandellion::seed_new_plant(int spawnpoint_x, int spawnpoint_y) {
    return new Dandellion(spawnpoint_x, spawnpoint_y, this->get_world());
}

void Dandellion::action() {
    for (int i = 0; i < DANDELLION_SEEDING_ATTEMPTS_PER_TURN; ++i) {
        this->seed();
    }
}
